client/src/entrypoints/admin/telepath/widgets.js

```diff
@@ -74,24 +74,29 @@ class Widget {
     /* write the HTML into a temp container to parse it into an element */
     const tempContainer = document.createElement('div');
     tempContainer.innerHTML = html.trim();
-    const dom = tempContainer.firstChild;
 
-    /* replace the placeholder with the new element */
-    placeholder.replaceWith(dom);
+    const selector = `:is(input,select,textarea,button)[name="${name}"]`;
+    // find, including element itself
+    const input = tempContainer.firstChild.matches(selector)
+      ? tempContainer.firstChild
+      : tempContainer.querySelector(selector);
```

Comment:
This logic is duplicated from the one in `BoundWidget`.

We could probably also remove the one in `BoundWidget`, but I'm not sure if this `render` method in the base `Widget` class is the only entrypoint where we render bound widgets.

We have a few widgets extending `Widget` but they only override `boundWidgetClass` without overriding `render()`, the only exception being `BaseDateTimeWidget` and that one doesn't even use `boundWidgetClass` (and the `render()` method signature doesn't have `parentCapabilities` and `options`).

So in terms of built-in widgets, removing this same logic in `BoundWidget` is probably safe, but I decided to leave it in. In theory, it should automatically pass the `element.matches(selector)` check and use the element as-is.

Reply:
Just noticed that even without removing it we already have a few failing test cases 🙈 .

Will look into it...

Reply:
OK, never mind all that. Found a better approach that doesn't change things too much.